rasa/core/channels/slack.py

```diff
@@ -399,7 +453,16 @@ async def health(_: Request) -> HTTPResponse:
         @slack_webhook.route("/webhook", methods=["GET", "POST"])
         async def webhook(request: Request) -> HTTPResponse:
             content_type = request.headers.get("content-type")
-            # Slack API sends either a JSON-encoded or a URL-encoded body depending on the content
+
+            if not self.is_request_from_slack_authentic(request):
```

Comment:
Is this the only channel that does message request level authentication/validation? And if not, is it always handled directly in the blueprint?

Reply:
I don't think all different channels have this functionality and some choose a different way (e.g. telegram recommends to choose an "unguessable" URL, e.g. https://www.example.com/<token>). We'll need to secure a couple more channels, it seems to be implemented for facebook, but I didn't yet check all of them

Reply:
ok, makes sense. Maybe when a few more are secured we could consider making a standard way of doing it e.g. an overridable abstract method on `InputChannel` that is called every time.